daemon/snap.go

```diff
@@ -227,15 +221,16 @@ func clientHealthFromHealthstate(h *healthstate.HealthState) *client.SnapHealth
 	}
 }
 
-func clientSnapRefreshInhibit(st *state.State, snapst *snapstate.SnapState, instanceName string) (*client.SnapRefreshInhibit, error) {
-	if !snapstate.IsSnapMonitored(st, instanceName) {
-		return nil, nil
+func clientSnapRefreshInhibit(st *state.State, snapst *snapstate.SnapState, instanceName string) *client.SnapRefreshInhibit {
+	proceedTime := snapst.RefreshInhibitProceedTime(st)
+	if (!proceedTime.IsZero() && proceedTime.Before(time.Now())) || !snapstate.IsSnapMonitored(st, instanceName) {
```

Comment:
as this is the negation of the OR condition we discussed I think || here needs to be &&, no?

Reply:
updated, thank you!